### ISSUE TITLE
Improve `on_load` events

### ISSUE THREAD

**Issue by picklelo** [pynecone-io/pynecone#603]
As a follow up to #596 we should rethink the design and potentially compile the on_load events directly into the page's initial event queue. This should allow them to follow the same flow as all other events.

**Comment by picklelo** [comment]
We should also make sure to handle async events:

```python
class State(...):
    async def onload(self): # was never awaited exception
        async with websockets.connect(...) as ws:
            ...
            recv = await ws.recv()
            self.data = recv
@pc.route("/", on_load=State.onload)
def index():
    return ...
```

**Comment by picklelo** [comment]
Merged in #773

**Comment by johanfforsberg** [comment]
I know this issue is closed, and perhaps I should open a new one, but I would like to comment that this kind of thing mentioned here still doesn't work AFAICT:

```python
class State(...):
    async def onload(self): # was never awaited exception
        async with websockets.connect(...) as ws:
            ...
            recv = await ws.recv()
            self.data = recv

@pc.route("/", on_load=State.onload)
def index():
    return ...
```
When I do something like this e.g. just
```python
    async def onload(self):
        i = 0
        while True:
            await asyncio.sleep(1)
            self.data = i
            i +=1
```
I get no updates. It seems nothing will happen until the `onload` function returns.

I can work around it by returning another event handler from `onload` and then let it return itself, to set up an infinite loop. But it seems more obscure and probably less efficient than just looping over an iterator.

pynecone 0.1.28
Python 3.10.6
